spec/grape/endpoint_spec.rb

```diff
@@ -298,6 +298,65 @@ def app
       end
     end
 
+    context 'when params are not built with default class' do
+      it 'should return an object that corresponds with the params class - hash with indifferent access' do
+        d = nil
+
+        subject.params do
+          build_with Grape::Extensions::ActiveSupport::HashWithIndifferentAccess::ParamBuilder
+
+          requires :first
+          optional :second, default: 'second-default'
+        end
+        subject.get '/declared' do
+          d = declared(params, include_missing: true)
+          ''
+        end
+
+        puts d
+        get '/declared?first=present'
+        expect(d.class).to eq(ActiveSupport::HashWithIndifferentAccess)
+      end
+
+      it 'should return an object that corresponds with the params class - hashie mash' do
+        d = nil
+
+        subject.params do
+          build_with Grape::Extensions::Hashie::Mash::ParamBuilder
+
+          requires :first
+          optional :second, default: 'second-default'
+        end
+        subject.get '/declared' do
+          d = declared(params, include_missing: true)
+          ''
+        end
+
+        puts d
+        get '/declared?first=present'
+        expect(d.class).to eq(Hashie::Mash)
+      end
+
+      it 'should return an object that corresponds with the params class - hash' do
+        d = nil
+
+        subject.params do
+          build_with Grape::Extensions::Hash::ParamBuilder
+
+          requires :first
+          optional :second, default: 'second-default'
+        end
+        subject.get '/declared' do
+          d = declared(params, include_missing: true)
```

Comment:
@dblock what is returned by the api call is always a json string. What we are testing here is what is returned by `declared`. Is there a better place to put this to test just that?